Add support for `Ctrl` + `<` / `>` to rotate selection in editor

peppy opened the pull request:
As discussed in https://github.com/ppy/osu/discussions/24048.

I've also added a visual effect when hotkeys are used to trigger the buttons for better visibility.

https://github.com/ppy/osu/assets/191335/7340a4c6-18b2-4b2f-9e9c-890e12a69ed0

peppy committed:
Add a visual effect when keyboard shortcuts are used to trigger selection box buttons

peppy committed:
Add support for `Ctrl` + `<` / `>` to rotate selection in editor

As discussed in https://github.com/ppy/osu/discussions/24048.

peppy commented on osu.Game/Screens/Edit/Compose/Components/SelectionBox.cs:
```diff
@@ -300,7 +301,7 @@ private void addYFlipComponents()
         }
 
-        private void addButton(IconUsage icon, string tooltip, Action action)
+        private SelectionBoxButton addButton(IconUsage icon, string tooltip, Action action)
```
Rather than returning the button and storing to fields, we could probably move the hotkey handling to inside the button. Maybe something for the future (although if it's seen as a better direction I will apply it here).

bdach replied:
I actually prefer this more than handling the key in the buttons themselves because I've tried handling hotkeys in buttons before on the mod overlay and it failed miserably as soon as the first hotkey conflict happened. So centralised is _probably_ better in this case.

bdach replied:
Also see https://github.com/ppy/osu/pull/24059/commits/9be2d9d62e131d22e98b650bdb82e48c97d8f18a - fixing that in the buttons would require further rejiggling.

bdach committed:
Enable NRT in `SelectionBox`

bdach committed:
Switch to `== true`

bdach committed:
Mention hotkeys in button tooltips

bdach committed:
Add test coverage for rotate hotkeys

bdach commented:
Follow-up changes applied:

- Enabled NRT on `SelectionBox` (https://github.com/ppy/osu/pull/24059/commits/ad3a470eafad8a22b678a45d6ff7a7da574764c4) - only did it since it was bothering me that the newly-added fields with button were not null-annotated
- Switched from `?? false` checks to `== true` (https://github.com/ppy/osu/pull/24059/commits/54280f06be5d8cb0dae7348061088278bbfd9a7e) - pretty sure we agreed on this style a while back now
- Added mention of hotkeys to tooltips (https://github.com/ppy/osu/pull/24059/commits/17ed45d07c2e8b6fba9995fc65a4963a613341ca) - mostly to match the reverse button tooltip
- Added test coverage (https://github.com/ppy/osu/pull/24059/commits/444f71541ad8894077674b7a350faf3fe7993c4f).

bdach committed:
Fix hotkey presses generating unnecessary undo history

The buttons don't check whether the operation they correspond to is
possible to perform in the current state of the selection box, so not
checking `Can{Reverse,Rotate}` causes superfluous undo states to be
added without any real changes if an attempt is made to reverse or
rotate a selection that cannot be reversed or rotated.
